tests/surface_input_regions.cpp

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wold <williamwold@canonical.com>
+ */
+
+#include "helpers.h"
+#include "in_process_server.h"
+
+#include <gmock/gmock.h>
+
+#include <vector>
+
+using namespace testing;
+
+// No tests use SUBTRACT_RECT as that is not yet implemented in Mir
+
+enum RegionAction
+{
+    ADD_RECT,
+    SUBTRACT_RECT
+};
```

Comment:
We'd normally use an `enum class` and CamelCase for the elements

Reply:
> We'd normally use an enum class and CamelCase for the elements

Actually our [guidelines](https://mir-server.io/doc/cppguide/index.html?showone=Enumerator_Names#Enumerator_Names) say lower_case for the members.